### PR TITLE
GH-103354: Replace Py_NewRef(Py_NotImplemented) usage with Py_RETURN_NOTIMPLEMENTED

### PR DESCRIPTION
**Pull request by furkanonder**
* Issue: gh-103354

### CODE REVIEW

**Comment by furkanonder** [comment]
I think we need to skip the news.

File: Modules/_xxinterpchannelsmodule.c [hunk 1]
```diff
@@ -1723,8 +1723,7 @@ channelid_richcompare(PyObject *self, PyObject *other, int op)
     }
 
     if (!PyObject_TypeCheck(self, state->ChannelIDType)) {
-        res = Py_NewRef(Py_NotImplemented);
-        goto done;
+        Py_RETURN_NOTIMPLEMENTED;
```
Comment (carljm) [inline Modules/_xxinterpchannelsmodule.c:1726]: These two uses are wrong, because we fail to decref `mod`.

**Review by carljm — COMMENTED** [review]
Thanks for the contribution, but this doesn't seem worth it to me. Of the five sites changed here, two cause a reference leak and two others disrupt the existing consistency of the code structure by adding early returns. And this change doesn't fix any actual problem; it seems like just a cosmetic change. AFAIK CPython policy is generally not to accept cosmetic-only PRs.

File: Modules/_xxinterpchannelsmodule.c (hunk 1 above)
Comment (sobolevn) [reply Modules/_xxinterpchannelsmodule.c:1726]: I agree, this is not the same thing. `#define Py_RETURN_NOTIMPLEMENTED return Py_NewRef(Py_NotImplemented)` while here we use `goto done`.

**Review by sobolevn — COMMENTED** [review]
I also think that it is not worth it + it has new issues (this is exactly why we don't like cosmetic changes: they have low value but can introduce new issues to existing code base).

I propose to close this PR.

But, thanks a lot for your contribution. I really like that you want to make CPython better, instead of this one - you can take any other existing issue and work on it. If you need help, feel free to ping me and ask your questions :)

**Pull request closed without merging**